### PR TITLE
Moving cargo check for runtimes to GHA

### PR DESCRIPTION
**Pull request by mutantcornholio**
Fixes https://github.com/paritytech/ci_cd/issues/1015

### CODE REVIEW

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+name: Check Cargo Check Runtimes
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
```
Comment (mordamax) [inline .github/workflows/check-cargo-check-runtimes.yml:5]:
```suggestion
    types: [ opened, synchronize, reopened, ready_for_review, labeled ]
```

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - id: set_image
+        run: cat .github/env >> $GITHUB_OUTPUT
+  check-runtime-assets:
+    runs-on: arc-runners-polkadot-sdk-beefy
+    needs: [set-image]
+    container:
+      image: docker.io/paritytech/ci-unified:bullseye-1.77.0-2024-04-10-v20240408
```
Comment (mordamax) [inline .github/workflows/check-cargo-check-runtimes.yml:29]:
```suggestion
      image: ${{ needs.set-image.outputs.IMAGE }}
```

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      IMAGE: ${{ steps.set_image.outputs.IMAGE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - id: set_image
+        run: cat .github/env >> $GITHUB_OUTPUT
+  check-runtime-assets:
+    runs-on: arc-runners-polkadot-sdk-beefy
+    needs: [set-image]
+    container:
```
Comment (alvicsam) [inline .github/workflows/check-cargo-check-runtimes.yml:28]: Please add timeout 20 min for each job.

**Review by alvicsam — CHANGES_REQUESTED** [review]
Good job! Small addition is still required (timeouts) and also I'd add a final job that depends on all other jobs (like [here](https://github.com/paritytech/polkadot-sdk/blob/master/.github/workflows/tests-linux-stable.yml#L115-L126)) in case we decide to mark these jobs `Required` in the future.

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      image: ${{ needs.set-image.outputs.IMAGE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run cargo check
+        uses: ./.github/actions/cargo-check-runtimes
+        with:
+          root: cumulus/parachains/runtimes/assets
+
+  check-runtime-collectives:
```
Comment (EgorPopelyaev) [inline .github/workflows/check-cargo-check-runtimes.yml:38]: How about to put the runtimes names in the matrix like:
matrix:
  runtimes: [assets, collectives, etc]
 to reduce amount of repetitive steps?

Comment (alvicsam) [reply .github/workflows/check-cargo-check-runtimes.yml:38]: We discussed that internally, there is [a comment](https://github.com/paritytech/polkadot-sdk/pull/5340/files#diff-ca6ddd9d3cdff8f696fe610915c9cd9316882794d8ed1e1e4dc6be392c0eb3e3R8). The main reason is that we have a certain amount of runners so using matrix uses them up faster. So instead running 9 jobs at once we run 2-3 jobs at one time and occupy less amount of runners.

Comment (EgorPopelyaev) [reply .github/workflows/check-cargo-check-runtimes.yml:38]: Oki, got it:)

File: .github/workflows/check-cargo-check-runtimes.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    outputs:
+      IMAGE: ${{ steps.set_image.outputs.IMAGE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - id: set_image
+        run: cat .github/env >> $GITHUB_OUTPUT
+  check-runtime-assets:
+    runs-on: arc-runners-polkadot-sdk-beefy
+    needs: [set-image]
```
Comment (alvicsam) [inline .github/workflows/check-cargo-check-runtimes.yml:27]:
```suggestion
    needs: [set-image]
    timeout-minutes: 20
```
last one :)

**Review by alvicsam — APPROVED** [review]

**Pull request merged**